src/error.ts

```diff
@@ -0,0 +1,53 @@
+import { STATUS_CODES } from 'node:http';
+import { format } from 'node:util';
+
+/**
+ * Base HTTP error class built with a message and a status code
+ * @extends Error
+ */
+export class ErrorHTTP extends Error {
+  status: number;
+
+  constructor(message?: string | number, status: number = 500) {
+    super();
+
+    if (!message) {
+      this.status = status;
+      this.message = STATUS_CODES[status] as string;
+    } else if (typeof message === 'number') {
+      this.status = message;
+      this.message = STATUS_CODES[this.status] as string;
+    } else {
+      this.message = message;
+      this.status = status;
+    }
+
+    Error.call(this, this.message);
+    Error.captureStackTrace(this, ErrorHTTP);
+  }
+}
+
+/**
+ * Generates a custom error object used for specific error handling.
+ * Extends the ErrorHTTP class.
+ *
+ * @param {String} code - error code you want to use, ex: RateLimitError
+ * @param {Number} status - the HTTP status code you want this error to throw
+ * @extends ErrorHTTP
+ * @returns {Class} - an error handling class
+ */
+export function fastErrorHTTP(code: string, status?: number) {
```

Comment:
question: where does the name fast come from here? I am mostly curious how it's faster than the class `ErrorHTTP`

Reply:
I was wondering the same thing - looks like it was related to a lib called "fasterror" and added in https://github.com/mapbox/mapbox-error/pull/13

Considering we're removing fastlog and have no reference to fasterror - I'm in favor of renaming this function to something more meaningful, like `customErrorHTTP`